src/net.cpp

```diff
@@ -1080,7 +1080,12 @@ void ThreadSocketHandler()
                 }
             }
         }
-        if(vNodes.size() != nPrevNodeCount) {
+        size_t vNodesSize;
+        {
+            LOCK(cs_vNodes);
+            vNodesSize = vNodes.size();
+        }
+        if(vNodesSize != nPrevNodeCount) {
             nPrevNodeCount = vNodes.size();
```

Comment:
@thelazier no, without `if` you would call `uiInterface.NotifyNumConnectionsChanged(nPrevNodeCount);` way too often

@spencerlievens reference patch in bitcoin has changes I'm talking about and your patch doesn't

bitcoin:
![screen shot 2017-05-07 at 14 42 43](https://cloud.githubusercontent.com/assets/1935069/25780642/96c60d96-3333-11e7-8c92-56a15488d3d4.png)

your patch:
![screen shot 2017-05-07 at 14 43 00](https://cloud.githubusercontent.com/assets/1935069/25780646/a7e8d75c-3333-11e7-866b-ebe2ba0c57f2.png)